runtime/src/snapshot_utils.rs

```diff
@@ -1538,30 +1535,52 @@ pub fn snapshot_bank(
     bank_snapshots_dir: impl AsRef<Path>,
     snapshot_archives_dir: impl AsRef<Path>,
     snapshot_version: SnapshotVersion,
-    archive_format: &ArchiveFormat,
+    archive_format: ArchiveFormat,
     hash_for_testing: Option<Hash>,
+    snapshot_type: Option<SnapshotType>,
 ) -> Result<()> {
-    let storages = root_bank.get_snapshot_storages(None);
+    let snapshot_storages = snapshot_type.map_or_else(SnapshotStorages::default, |snapshot_type| {
+        let incremental_snapshot_base_slot = match snapshot_type {
+            SnapshotType::IncrementalSnapshot(incremental_snapshot_base_slot) => {
+                Some(incremental_snapshot_base_slot)
+            }
+            _ => None,
+        };
+        root_bank.get_snapshot_storages(incremental_snapshot_base_slot)
+    });
     let mut add_snapshot_time = Measure::start("add-snapshot-ms");
-    add_bank_snapshot(&bank_snapshots_dir, root_bank, &storages, snapshot_version)?;
+    add_bank_snapshot(
+        &bank_snapshots_dir,
+        root_bank,
+        &snapshot_storages,
+        snapshot_version,
+    )
+    .expect("failed to serialize bank snapshot to disk");
```

Comment:
should we add some safeguards to make sure `add_bank_snapshot` *only* returns nonretryable errors? It seems like `add_bank_snapshot` returns type `SnapshotError`. Right now it seems all of them are fatal, but should we introduce a `is_fatal()` function on `SnapshotError` in case somebody adds a new variant to it? Then we can say 

```
if let Err(e) = add_bank_snapshot {
       if e.is_fatal() {
           panic!("failed to serialize bank snapshot to disk");
       }
}
```

Reply:
Handled offline.